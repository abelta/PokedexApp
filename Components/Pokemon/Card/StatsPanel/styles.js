import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  statsPanel: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});

export default styles;
