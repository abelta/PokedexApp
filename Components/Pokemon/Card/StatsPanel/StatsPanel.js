import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { pokemonStatShape } from '../../../../PropTypes/Pokemon';
import { StatColumn } from '../.';
import styles from './styles';

const StatsPanel = ({ data }) => (
  <View style={styles.statsPanel}>
    {
      data.map(stat => (
        <StatColumn key={stat.stat.name} label={stat.stat.name} value={stat.baseStat} />
      ))
    }
  </View>
);

StatsPanel.propTypes = {
  data: PropTypes.arrayOf(pokemonStatShape).isRequired,
};

export default StatsPanel;
