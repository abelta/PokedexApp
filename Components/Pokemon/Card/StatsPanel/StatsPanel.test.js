import React from 'react';
import renderer from 'react-test-renderer';
import camelcaseKeys from 'camelcase-keys';
import StatsPanel from '.';
import data from './mock-data.json';

describe('StatsPanel', () => {
  it('renders properly', () => {
    const tree = renderer.create(
      <StatsPanel data={camelcaseKeys(data, { deep: true }).stats} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
