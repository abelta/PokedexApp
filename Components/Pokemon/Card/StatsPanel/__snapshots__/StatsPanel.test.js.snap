// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsPanel renders properly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "baseline",
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "space-between",
        "marginTop": 10,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "height": 120,
        "marginLeft": 5,
        "marginRight": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "bottom": 20,
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "backgroundColor": "pink",
            "height": 100,
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "lightgrey",
          "bottom": 0,
          "fontSize": 12,
          "height": 20,
          "position": "absolute",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Speed
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "height": 120,
        "marginLeft": 5,
        "marginRight": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "bottom": 20,
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "backgroundColor": "lightgreen",
            "height": 85,
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "lightgrey",
          "bottom": 0,
          "fontSize": 12,
          "height": 20,
          "position": "absolute",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Sp. Def
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "height": 120,
        "marginLeft": 5,
        "marginRight": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "bottom": 20,
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "backgroundColor": "lightblue",
            "height": 109,
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "lightgrey",
          "bottom": 0,
          "fontSize": 12,
          "height": 20,
          "position": "absolute",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Sp. Atk
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "height": 120,
        "marginLeft": 5,
        "marginRight": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "bottom": 20,
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "backgroundColor": "yellow",
            "height": 78,
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "lightgrey",
          "bottom": 0,
          "fontSize": 12,
          "height": 20,
          "position": "absolute",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Def
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "height": 120,
        "marginLeft": 5,
        "marginRight": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "bottom": 20,
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "backgroundColor": "orange",
            "height": 84,
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "lightgrey",
          "bottom": 0,
          "fontSize": 12,
          "height": 20,
          "position": "absolute",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Atk
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "height": 120,
        "marginLeft": 5,
        "marginRight": 5,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "bottom": 20,
            "position": "absolute",
            "width": "100%",
          },
          Object {
            "backgroundColor": "red",
            "height": 78,
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "lightgrey",
          "bottom": 0,
          "fontSize": 12,
          "height": 20,
          "position": "absolute",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      HP
    </Text>
  </View>
</View>
`;
